includes/forms/class-wp-job-manager-form-submit-job.php

```diff
@@ -1209,4 +1211,75 @@ public function track_job_submission( $post_id ) {
 			]
 		);
 	}
+
+	/**
+	 * Checks if 'extend' is set as action.
+	 *
+	 * @since $$next-version$$
+	 * @return bool
+	 */
+	public function is_extend_action() {
+		return isset( $_GET['action'] ) && sanitize_text_field( wp_unslash( $_GET['action'] ) ) === 'extend';
+	}
+
+	/**
+	 * Handle steps for extending a listing's expiry.
+	 *
+	 * @access private
+	 * @param array $steps
+	 * @return array
+	 */
+	public function maybe_extend_job_expiry( $steps ) {
+		if ( $this->is_extend_action() ) {
+			unset( $steps['submit'] );
+			unset( $steps['preview'] );
+			$steps['extend-expiry'] = [
+				'name'     => 'Extend Expiry',
+				'view'     => [ $this, 'preview' ],
+				'handler'  => [ $this, 'extend_preview_handler' ],
+				'priority' => 26,
+			];
+		}
+		return $steps;
+	}
+
+	/**
+	 * Handles the extend-listing step.
+	 *
+	 * @throws Exception On validation error.
+	 */
+	public function extend_preview_handler() {
+		// phpcs:ignore WordPress.Security.NonceVerification.Missing -- Input is used safely.
+		if ( empty( $_POST ) ) {
+			return;
+		}
+
+		$this->check_preview_form_nonce_field();
+
+		if ( ! empty( $_POST['continue'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Missing -- Input is used safely.
+			$job       = get_post( $this->job_id );
+
+			$extending = $this->is_extend_action() && job_manager_job_can_be_extended( $this->job_id );
+
+			if ( $extending ) {
+				$old_expiry = date_create_immutable_from_format( 'Y-m-d', get_post_meta( $job->ID, '_job_expires', true ) );
+				$new_expiry = calculate_job_expiry( $job->ID, false, $old_expiry );
+				update_post_meta( $job->ID, '_job_expires', $new_expiry );
+			}
+
+			$requires_approval = get_option( 'job_manager_submission_requires_approval' );
+			$post_status       = $extending || ! $requires_approval ? 'publish' : 'pending';
```

Comment:
Can you explain this part a bit? The way that I understand the code $extending is always going to be true.

So AFAIU the only way `extend_preview_handler` is going to be called is if the user clicked the 'Extend' link which only appears if `job_manager_job_can_be_extended` is true. And we also call `is_extend_action` when we add the handler in the first place. 

Am I missing something?

Reply:
So I am thinking that generally, the admin approval is required to make sure that no spammy content is published. As long as we don't allow editing as part of this flow, it should be ok to simply republish it. Do you think of any scenarios that renewing content that was approved before might need this extra approval?